attestation-service/docs/amd-offline-certificate-cache.md

```diff
@@ -0,0 +1,246 @@
+without connection to the AMD KDS. Trustee supports certificate caching through
+the rust `http_cache_reqwest` crate, which caches URL responses into a
+specified folder that can be preloaded with certificates using the
+[cache-preloader](../../tools/cache-preloader) tool. This use case is primarily
+targeted at air-gapped environments as a stopgap solution while a more
+comprehensive approach is developed.
+
+> [!Note]
+> Currently this guide shows a method that builds on using
+> [docker](https://confidentialcontainers.org/docs/attestation/installation/docker/)
+> to manage kbs services. Information about enabling it with the
+> [Trustee Operator](https://confidentialcontainers.org/docs/attestation/installation/kubernetes/)
+> will be added in the near future.
+
+## Contents
+
+- [Prerequisites](#prerequisites)
+- [Method for Cache Population](#method-for-cache-population)
+- [Understanding the Cache Structure](#understanding-the-cache-structure)
+- [Limitations](#limitations)
+
+## Prerequisites
+
+- On system used to build cache:
+  - Network access to AMD KDS (https://kdsintf.amd.com)
+  - Rust toolchain installed (for building cache-preloader)
+- On AMD EPYC host(s) with guests to be attested:
+  - snphost tool ([link](https://github.com/virtee/snphost))
+
+## Method for Cache Population
+
+### 1. Create URL target file
+
+Create a text file with the list of URLs that you would like cached. The KDS
+will need a VCEK URL for each physical AMD EPYC server that the KDS will be
+servicing. The machine's VCEK URL can be fetched using the
+[snphost tool](https://github.com/virtee/snphost).
+
+On each AMD SNP host, with root/admin access run:
+
+```
+snphost show vcek-url
+```
+
+You should get a URL of the form:
+
+```
+https://kdsintf.amd.com/vcek/{version}/{machine}/{product_name}/{hwid}?{params}
+```
+
+See the [AMD VCEK documentation](https://www.amd.com/content/dam/amd/en/documents/epyc-technical-docs/specifications/57230.pdf)
+for more information about the VCEK URL format.
+
+> [!IMPORTANT]
+> Note that the VCEK URL is specific to the hardware AMD firmware of the
+> machine. If the firmware is updated, the VCEK URL will change.
+
+### 2. Run cache_preloader to generate cache contents
+
+On a system with access to the AMD KDS, run `cache_preloader` specifying
+the VCEK URL file and output directory.
+
+```
+# Navigate to the cache-preloader directory
+cd tools/cache-preloader
+
+# Preload the cache and create an archive
+cargo run -- -u vcek_urls.txt -c ./vcek_cache -a vcek_cache.tar.gz
+```
+
+### 3. Configure the attestation service
+
+Update the attestation config file to use a predefined cache:
+
+`kbs/config/as-config.json`:
+
+```json
+{
+    // ... other fields ...
+    "verifier_config": {
+        "snp_verifier": {
+            "cache": {
+                "type": "HttpCacheReqwest",
+                "offline_mode": true,
+                "cache_dir": "/path/to/cache"
+            }
```

Comment:
following my comment in the other cache PR: could we try to make the cache (config) design such that it's not TEE specific. I could retrofit TDX/SGX CollateralService setup under the same Trustee configs

the default for `cache_dir` should probably be aligned with the base path discussion (#992) . 

need any cache bust settings to check for fresh content every now and then?